ducktape/tests/test.py

```diff
@@ -380,3 +401,16 @@ def logger(self):
                 self.logger_name, self.results_dir, self.session_context.debug)
         return self._logger
 
+    def close(self):
+        """Release resources, etc."""
+        # Remove reference to services. This is important to prevent potential memory leaks if users write services
+        # which themselves have references to large memory-intensive objects
+        del self.services
+
+        # Remove local scratch directory
+        if self._scratch_dir and os.path.exists(self.scratch_dir):
```

Comment:
Mixing `_scratch_dir` and `scratch_dir`? If this needs to know about the internal field and that it may be `None`, just use it consistently.